hadoop-crypto/src/test/java/com/palantir/hadoop/cipher/AbstractFsCipherInputStreamTest.java

```diff
@@ -165,8 +166,11 @@ public final void testBulkRead() throws IOException {
         byte[] buffer = new byte[NUM_BYTES];
         int offset = 0;
         int read;
-        while ((read = cis.read(buffer, offset, buffer.length)) != -1) {
+        int len = buffer.length;
+        while ((read = cis.read(buffer, offset, len)) != 0) {
             offset += read;
+            len -= read;
```

Comment:
I made the change because the test was failing with an `IndexOutOfBoundsException` from https://github.com/apache/commons-crypto/blob/master/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java#L244. We were hitting the `len > b.length - off` case that it checks for. Is there a better way to handle that?